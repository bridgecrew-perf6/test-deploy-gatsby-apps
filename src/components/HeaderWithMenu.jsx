import React from "react"
import {
  AppBar,
  Button,
  Grid,
  makeStyles,
  Toolbar,
  Typography,
} from "@material-ui/core"
import cart from "../images/cart.svg"
import { Link } from "gatsby"
import { MuktiLogo } from "./shared/MuktiLogo"

const useStyles = makeStyles(theme => ({
  toolBar: {
    flexGrow: "1",
    display: "flex",
    justifyContent: "flex-end",
  },
  route: {
    textDecoration: "none",
    marginRight: "4vw",
  },
  routeButton: { color: theme.palette.secondary.main, borderRadius: "20px" },
  cartImg: { height: "4vh", width: "4vw", cursor: "pointer" },
}))

export const HeaderWithMenu = ({ routes, cartSize }) => {
  const classes = useStyles()
  return (
    <div style={{ flexGrow: "1" }}>
      <AppBar
        position="fixed"
        style={{
          boxShadow: "0px 0px 0px 0px",
        }}
      >
        <Grid container>
          <MuktiLogo />
          <Toolbar className={classes.toolBar}>
            {routes.map((route, key) => {
              return (
                <Link to={route.path} key={key} className={classes.route}>
                  <Button className={classes.routeButton}>{route.name}</Button>
                </Link>
              )
            })}
            <Link to={"/cart"} className={classes.route}>
              <Button className={classes.routeButton}>
                <img src={cart} className={classes.cartImg} alt={"cart"} />
                <Typography>{cartSize || ""}</Typography>
              </Button>
            </Link>
          </Toolbar>
        </Grid>
      </AppBar>
    </div>
  )
}
