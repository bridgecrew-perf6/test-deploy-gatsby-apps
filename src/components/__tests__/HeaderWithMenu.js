import React from "react"
import renderer from "react-test-renderer"
import { HeaderWithMenu } from "../HeaderWithMenu"
import { render, screen } from "@testing-library/react"

describe("HeaderWithMenu Snapshot", function () {
  it("renders HeaderWithMenu correctly", () => {
    const tree = renderer.create(<HeaderWithMenu routes={[]} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe("renders header components with menu", () => {
  it("renders 2 buttons with no routes", () => {
    render(<HeaderWithMenu routes={[]} />)
    expect(screen.getAllByRole("button")).toHaveLength(2)
  })

  it("renders 4 buttons with all routes", () => {
    const routes = [
      { name: "Food & Drinks", route: "/menu" },
      { name: "about", route: "/about" },
    ]
    render(<HeaderWithMenu routes={routes} />)
    expect(screen.getAllByRole("button")).toHaveLength(4)
  })

  it("renders Mukti logo image", () => {
    render(<HeaderWithMenu routes={[]} />)
    const logo = screen.getByAltText("logo")
    expect(logo).toBeInTheDocument()
  })

  it("renders cart image", () => {
    render(<HeaderWithMenu routes={[]} />)
    const logo = screen.getByAltText("cart")
    expect(logo).toBeInTheDocument()
  })

  it("renders cart-size for non-empty cart", () => {
    render(<HeaderWithMenu routes={[]} cartSize={"5"} />)
    const cartSizeText = screen.getByText("5")
    expect(cartSizeText).toBeInTheDocument()
  })

  it("render menu text", () => {
    const routes = [
      { name: "Food & Drinks", route: "/menu" },
      { name: "About", route: "/about" },
    ]
    render(<HeaderWithMenu routes={routes} />)
    const menuText = screen.getByText("Food & Drinks")
    const aboutText = screen.getByText("About")
    expect(menuText).toBeInTheDocument()
    expect(aboutText).toBeInTheDocument()
  })
})
