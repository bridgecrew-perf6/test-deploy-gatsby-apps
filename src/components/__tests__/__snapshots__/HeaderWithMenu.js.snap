// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderWithMenu Snapshot renders HeaderWithMenu correctly 1`] = `
<div
  data-cy="with-menu"
  style={
    Object {
      "flexGrow": "1",
    }
  }
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    style={
      Object {
        "boxShadow": "0px 0px 0px 0px",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-5"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <img
            alt="logo"
            className="makeStyles-logo-6"
            src="test-file-stub"
          />
        </span>
      </button>
      <div
        className="MuiToolbar-root MuiToolbar-regular makeStyles-toolBar-1 MuiToolbar-gutters"
      >
        <a
          className="makeStyles-route-2"
          href="/cart"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-routeButton-3"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <img
                alt="cart"
                className="makeStyles-cartImg-4"
                src="test-file-stub"
              />
              <p
                className="MuiTypography-root MuiTypography-body1"
                data-cy="cart-size"
              >
                
              </p>
            </span>
          </button>
        </a>
      </div>
    </div>
  </header>
</div>
`;
